src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -218,10 +249,12 @@ class HuiMapCard extends LitElement implements LovelaceCard {
       ratio && ratio.w > 0 && ratio.h > 0
         ? `${((100 * ratio.h) / ratio.w).toFixed(2)}%`
         : (root.style.paddingBottom = "100%");
+
+    this._date = new Date();
   }
 
   protected updated(changedProps: PropertyValues): void {
-    if (changedProps.has("hass")) {
+    if (changedProps.has("hass") || changedProps.has("_history")) {
       this._drawEntities();
       this._fitMap();
```

Comment:
This is existing code, but isn't it a bad experience that we keep fitting the map on a state change? It won't allow users to explore data, especially now that we're showing historic data.

Reply:
https://developers.google.com/maps/documentation/javascript/examples/control-custom

Probably for another PR though

Reply:
@zsarnett this is not google maps

Reply:
balloob is right. We center the map on every hass change (and now history change), to fit all markers. But this is bad user experience, when user want to explore the historic data. 

In an extra PR I can add a 'center map' button to refit the map. And a inactivity tracker to refit the map would be useful, too.